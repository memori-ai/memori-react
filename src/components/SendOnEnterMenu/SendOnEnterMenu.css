.memori-send-on-enter-menu {
    position: relative;
    display: inline-block;
}

.memori--conversation-button {
    cursor: pointer;
}

.memori-menu--overlay {
    position: absolute;
    right: auto;
    bottom: 100%;
    left: 0;
    display: flex;
    min-width: 12rem;
    flex-direction: column;
    padding: 0.5rem 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
    border-radius: 0.375rem;
    margin: 0;
    margin-top: 0.5rem;
    background-color: #fff;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.12);
    color: #000;
    list-style: none;
    outline: none;
    transform-origin: top right;
}

.memori-menu--option {
    display: flex;
    align-items: center;
    justify-content: center;
}

.memori-menu--button {
    width: 100%;
    justify-content: flex-start;
    color: inherit;
    text-align: left;
    text-decoration: none;
}

.memori-menu--icon {
    width: 1em;
}
