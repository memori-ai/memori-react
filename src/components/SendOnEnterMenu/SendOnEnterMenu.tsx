import React from 'react';
import { Menu, RadioGroup } from '@headlessui/react';
import cx from 'classnames';
import { useTranslation } from 'react-i18next';
import Button from '../ui/Button';

export interface Props {
  sendOnEnter: 'keypress' | 'click';
  setSendOnEnter: (value: 'keypress' | 'click') => void;
}

const SendOnEnterMenu: React.FC<Props> = ({ sendOnEnter, setSendOnEnter }) => {
  const { t } = useTranslation();

  return (
    <Menu as="div" className="memori-send-on-enter-menu">
      <Menu.Button
        className={cx(
          'memori-button',
          'memori-button--circle',
          'memori-button--icon-only',
          'memori-share-button--button',
          'memori--conversation-button'
        )}
      >
        <div className="memori-button--icon">
          <span
            style={{
              display: 'block',
              width: '1rem',
              height: '1rem',
            }}
          >
            …
          </span>
        </div>
      </Menu.Button>
      <Menu.Items className="memori-menu--overlay">
        <RadioGroup value={sendOnEnter} onChange={setSendOnEnter}>
          <RadioGroup.Option value="keypress" className="memori-menu--option">
            {({ checked }) => (
              <Menu.Item>
                <Button
                  className="memori-menu--button"
                  ghost
                  outlined={checked}
                  icon={
                    <span className="memori-menu--icon">
                      {checked ? '✓' : ''}
                    </span>
                  }
                >
                  {t('widget.sendOnKeypress')}
                </Button>
              </Menu.Item>
            )}
          </RadioGroup.Option>
          <RadioGroup.Option value="click" className="memori-menu--option">
            {({ checked }) => (
              <Menu.Item>
                <Button
                  className="memori-menu--button"
                  ghost
                  outlined={checked}
                  icon={
                    <span className="memori-menu--icon">
                      {checked ? '✓' : ''}
                    </span>
                  }
                >
                  {t('widget.sendOnClick')}
                </Button>
              </Menu.Item>
            )}
          </RadioGroup.Option>
        </RadioGroup>
      </Menu.Items>
    </Menu>
  );
};

export default SendOnEnterMenu;
