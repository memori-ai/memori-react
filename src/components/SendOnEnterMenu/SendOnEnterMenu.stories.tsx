import React from 'react';
import { Meta, Story } from '@storybook/react';
import I18nWrapper from '../../I18nWrapper';
import SendOnEnterMenu, { Props } from './SendOnEnterMenu';

import './SendOnEnterMenu.css';

const meta: Meta = {
  title: 'Widget/Send On Enter Menu',
  component: SendOnEnterMenu,
  argTypes: {
    sendOnEnter: {
      control: {
        type: 'select',
        options: ['keypress', 'click'],
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<Props> = args => {
  const [sendOnEnter, setSendOnEnter] = React.useState(args.sendOnEnter);

  return (
    <I18nWrapper>
      <div
        style={{
          minHeight: '200px',
          display: 'flex',
          alignItems: 'flex-end',
        }}
      >
        <SendOnEnterMenu
          {...args}
          sendOnEnter={sendOnEnter}
          setSendOnEnter={setSendOnEnter}
        />
      </div>
    </I18nWrapper>
  );
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Click = Template.bind({});
Click.args = {
  sendOnEnter: 'click',
};

export const Keypress = Template.bind({});
Keypress.args = {
  sendOnEnter: 'keypress',
};
