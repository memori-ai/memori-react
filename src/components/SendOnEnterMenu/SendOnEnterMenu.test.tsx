import React from 'react';
import { render } from '@testing-library/react';
import SendOnEnterMenu from './SendOnEnterMenu';

it('renders SendOnEnterMenu on click unchanged', () => {
  const { container } = render(
    <SendOnEnterMenu sendOnEnter="click" setSendOnEnter={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});

it('renders SendOnEnterMenu on keypress unchanged', () => {
  const { container } = render(
    <SendOnEnterMenu sendOnEnter="keypress" setSendOnEnter={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});
