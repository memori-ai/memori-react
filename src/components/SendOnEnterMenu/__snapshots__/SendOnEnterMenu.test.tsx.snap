// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SendOnEnterMenu on click unchanged 1`] = `
<div>
  <div
    class="memori-send-on-enter-menu"
    data-headlessui-state=""
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="memori-button memori-button--circle memori-button--icon-only memori-share-button--button memori--conversation-button"
      data-headlessui-state=""
      id="headlessui-menu-button-:r0:"
      type="button"
    >
      <div
        class="memori-button--icon"
      >
        <span
          style="display: block; width: 1rem; height: 1rem;"
        >
          …
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`renders SendOnEnterMenu on keypress unchanged 1`] = `
<div>
  <div
    class="memori-send-on-enter-menu"
    data-headlessui-state=""
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="memori-button memori-button--circle memori-button--icon-only memori-share-button--button memori--conversation-button"
      data-headlessui-state=""
      id="headlessui-menu-button-:r2:"
      type="button"
    >
      <div
        class="memori-button--icon"
      >
        <span
          style="display: block; width: 1rem; height: 1rem;"
        >
          …
        </span>
      </div>
    </button>
  </div>
</div>
`;
